### PR TITLE
chore(integrations): move cassandra,celery,cherrypy,consul,coverage to internal

### PR DESCRIPTION
**Pull request by rachelyangdog**
- Moves all integration internals in ddtrace/contrib/(integration name)/ to ddtrace/contrib/internal/(integration name)/ for cassandra, celery, cherrypy, consul, and coverage
- Ensures ddtrace/contrib/(integration name)/ and ddtrace/contrib/(integration name)/ continue to expose the same functions, classes, imports, and module level variables (via from ..internal.integration.module import * imports).
- Log a deprecation warning if internal modules in ddtrace/contrib/(integration name)/ and ddtrace/contrib/(integration name)/. Only patch and unpack methods should be exposed by these packages.
- https://github.com/DataDog/dd-trace-py/pull/9996

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by mabdinur — COMMENTED** [review]
Can we update the PR description

File: ddtrace/contrib/internal/cassandra/patch.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from ddtrace.contrib.internal.cassandra.session import patch
+from ddtrace.contrib.internal.cassandra.session import unpatch
+
+
+__all__ = ["patch", "unpatch"]
```
Comment (mabdinur) [inline ddtrace/contrib/internal/cassandra/patch.py:5]: Do we need this file? If cassandra doesn't have a patch module, I don't think we should add it.

File: ddtrace/contrib/cassandra/__init__.py [hunk 2]
```diff
@@ -28,7 +28,11 @@
 
 with require_modules(required_modules) as missing_modules:
     if not missing_modules:
-        from .patch import patch
-        from .session import get_version
+        # Required to allow users to import from `ddtrace.contrib.cassandra.patch` directly
+        from . import patch as _  # noqa: F401, I001
```
Comment (mabdinur) [inline ddtrace/contrib/cassandra/__init__.py:32]:
```suggestion
```

File: ddtrace/contrib/cassandra/__init__.py [hunk 3]
```diff
@@ -28,7 +28,11 @@
...
+
+        # Expose public methods
+        from ..internal.cassandra.patch import patch
```
Comment (mabdinur) [inline ddtrace/contrib/cassandra/__init__.py:35]:
```suggestion
        from ..internal.cassandra.session import patch
```

File: ddtrace/contrib/cassandra/__init__.py [hunk 4]
```diff
@@ -28,7 +28,11 @@
 
 with require_modules(required_modules) as missing_modules:
     if not missing_modules:
-        from .patch import patch
-        from .session import get_version
+        # Required to allow users to import from `ddtrace.contrib.cassandra.patch` directly
```
Comment (mabdinur) [inline ddtrace/contrib/cassandra/__init__.py:31]:
```suggestion
```

File: ddtrace/contrib/internal/cassandra/patch.py (hunk 1 above)
Comment (mabdinur) [inline ddtrace/contrib/internal/cassandra/patch.py:5]:
```suggestion
```

File: ddtrace/contrib/cassandra/__init__.py [hunk 5]
```diff
@@ -28,7 +28,9 @@
 
 with require_modules(required_modules) as missing_modules:
```
Comment (mabdinur) [inline ddtrace/contrib/cassandra/__init__.py:29]: I think we accidentally deleted the `ddtrace/contrib/cassandra/patch.py` file: https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/contrib/cassandra/patch.py. Can we add it back and follow the deprecation pattern we used in other integration this should fix the failing test

File: tests/contrib/cherrypy/test_middleware.py [hunk 6]
```diff
@@ -55,7 +55,7 @@ def setUp(self):
         )
 
     def test_and_emit_get_version(self):
-        from ddtrace.contrib.cherrypy import get_version
+        from ddtrace.contrib.cherrypy.middleware import get_version
```
Comment (mabdinur) [inline tests/contrib/cherrypy/test_middleware.py:58]:
```suggestion
        from ddtrace.contrib.cherrypy import get_version
```

File: tests/contrib/cherrypy/test_middleware.py [hunk 7]
```diff
@@ -15,7 +15,7 @@
 from ddtrace.constants import ERROR_STACK
 from ddtrace.constants import ERROR_TYPE
 from ddtrace.constants import SAMPLING_PRIORITY_KEY
-from ddtrace.contrib.cherrypy import TraceMiddleware
+from ddtrace.contrib.internal.cherrypy.middleware import TraceMiddleware
```
Comment (mabdinur) [inline tests/contrib/cherrypy/test_middleware.py:18]:
```suggestion
from ddtrace.contrib.cherrypy import TraceMiddleware
```

File: tests/contrib/cassandra/test_cassandra_patch.py [hunk 8]
```diff
@@ -4,7 +4,7 @@
 # from being overwritten by future re-generations.
 
 from ddtrace.contrib.cassandra import get_version
-from ddtrace.contrib.cassandra.patch import patch
+from ddtrace.contrib.internal.cassandra.patch import patch
```
Comment (mabdinur) [inline tests/contrib/cassandra/test_cassandra_patch.py:7]:
```suggestion
from ddtrace.contrib.cassandra.patch import patch
```

File: tests/contrib/cassandra/test.py [hunk 9]
```diff
@@ -16,7 +16,7 @@
 from ddtrace.constants import ERROR_TYPE
 from ddtrace.contrib.cassandra.patch import patch
 from ddtrace.contrib.cassandra.patch import unpatch
-from ddtrace.contrib.cassandra.session import SERVICE
+from ddtrace.contrib.internal.cassandra.session import SERVICE
```
Comment (mabdinur) [inline tests/contrib/cassandra/test.py:19]:
```suggestion
from ddtrace.contrib.cassandra.session import SERVICE
```

File: releasenotes/notes/move-integrations-to-internal-cassandra-c81f9627c3c7ed2c.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+---
+deprecations:
+  - |
+    cassandra: Deprecates all modules in the ddtrace.contrib.cassandra package. Use attributes exposed in ddtrace.contrib.cassandra.__all__ instead.
```
Comment (emmettbutler) [inline releasenotes/notes/move-integrations-to-internal-cassandra-c81f9627c3c7ed2c.yaml:4]:
```suggestion
    cassandra: Deprecates all modules in the ``ddtrace.contrib.cassandra`` package. Use attributes exposed in ``ddtrace.contrib.cassandra.__all__`` instead.
```

**Review by emmettbutler — APPROVED** [review]

**Comment by rachelyangdog** [comment]
@DataDog/ci-app-libraries Could I get a review please. Thank you!!

**Review by mabdinur — APPROVED** [review]

**Review by wconti27 — APPROVED** [review]

**Review by romainkomorndatadog — APPROVED** [review]

**Pull request merged**